### DISCUSSION TITLE
lz4 decompress

### DISCUSSION THREAD

**Discussion by pizza666** [cc65/cc65#1737] [Q&A]
I want to use lz4 for my projects but cant get the compressed data correctly compressed to decompress it with decompress_lz4 from https://github.com/cc65/cc65/blob/master/include/lz4.h

The closest results I got using lz4 https://github.com/lz4/lz4 was with the mentioned level 16 in the docs:
```
lz4 -l -16 src dest
```

The compressed result has some leading and trailing trash:
![image](https://user-images.githubusercontent.com/28303235/166110272-57b013a9-13ca-4865-8734-3a62ae0998a7.png)
Loading uncompressed file (or how the unit test should look like):
![image](https://user-images.githubusercontent.com/28303235/166110436-0c2adeea-0bc0-46c5-98c1-569a58494786.png)

Any advice?

**Comment by pizza666** [comment]
I got the correct output by skipping 8bytes of the compressed file.
Ill guess this is part of the frame header described here:
https://android.googlesource.com/platform/external/lz4/+/HEAD/doc/lz4_Frame_format.md

**Comment by mrdudz** [comment]
I wouldnt mind if you could provide a comple sample program that demonstrates how to use this :)

**Reply by pizza666** [reply]
Sure pushed it "quick and dirty" here:
https://github.com/pizza666/lz

**Comment by clbr** [comment]
The function corresponds to the similar function in the lz4 library - that is, it takes the output of LZ4_compress() or LZ4_compress_HC() from the lib. The lz4 command line tool uses a wrapped format to support larger files, and is not suitable for this.
